### ISSUE TITLE
Rector runs into errors after update to 2.6.0

### ISSUE THREAD

**Issue by steffenf14** [martin-helmich/typo3-typoscript-parser#100]
TYPO3 v11 LTS
Rector 0.17.0
PHP 8.2

  - Upgrading helmich/typo3-typoscript-parser (v2.5.0 => v2.6.0): Extracting archive
  - Upgrading helmich/typo3-typoscript-lint (v3.1.0 => v3.2.0): Extracting archive

after updating from 2.5.0 to 2.6.0 rector runs into this error:
```
./app/vendor/bin/rector -c build/config/rector.php --dry-run process

 [ERROR] Cannot autowire service "Helmich\TypoScriptParser\Parser\Traverser\Traverser": argument "$statements" of method
         "__construct()" is type-hinted "array", you should configure its value explicitly.

Failed to execute command ./app/vendor/bin/rector -c build/config/rector.php --dry-run process: exit status 1
```

**Comment by wtfred** [comment]
Hello, same problem here, dunno if this actually comes from here, or another package.
Downgrade to 2.5 fix it:

```
>>> composer req helmich/typo3-typoscript-parser:~2.5.0 -W
./composer.json has been updated
Running composer update helmich/typo3-typoscript-parser --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 4 updates, 0 removals
  - Downgrading helmich/typo3-typoscript-parser (v2.6.0 => v2.5.0)
  - Downgrading symfony/config (v7.0.4 => v6.4.4)
  - Downgrading symfony/dependency-injection (v7.0.4 => v6.4.4)
  - Downgrading symfony/yaml (v7.0.3 => v6.4.3)
  ...
  ```

**Comment by andreaswolf** [comment]
I'm pretty sure this is related to some change in 2.6, could also fix it by downgrading to 2.5.

**Comment by martin-helmich** [comment]
I'm unsure what's happening here. The signature of the `Helmich\TypoScriptParser\Parser\Traverser\Traverser` constructor hasn't changed since nearly 10 years (goes all the way back to the initial commit https://github.com/martin-helmich/typo3-typoscript-parser/commit/a90b4771cc9eb58f397480a2094d0dbbb8745549 from 2014). The only thing that's changed is the `@param` annotation and the (private) property types... 🤔 

I don't really know anything about Rector, or what it's trying to do, here (maybe initialising a Traverser via some kind of DI container?). Maybe someone could give me some hints on how a minimal, reproducible example might look like?

**Comment by sabbelasichon** [comment]
I will have a look at it

**Comment by martin-helmich** [comment]
@andreaswolf Possibly; I've prepared #101, which changes the annotation to `@param` (I've been out of the PHP world for a while and forgot how much tooling actually relies on those doc comments 😕).

Can you give it a test run by requiring the `dev-bugfix-param-annotation` version of this library?

**Comment by sabbelasichon** [comment]
As a quick fix you can remove the Traverser from the Symfony DI in your config

```php
$services->remove(\Helmich\TypoScriptParser\Parser\Traverser\Traverser::class)
```

**Comment by sabbelasichon** [comment]
The new release 1.8.1 should fix this error https://github.com/sabbelasichon/typo3-rector/releases/tag/v1.8.1

**Comment by sabbelasichon** [comment]
@martin-helmich Feel free to close this issue.

**Comment by magicsunday** [comment]
If you update the constructor of `\Helmich\TypoScriptParser\Parser\Traverser\Traverser` from:

```
    /**
     * @psalm-param Statement[] $statements
     */
    public function __construct(array $statements)
```

to 

```
    /**
     * @param Statement[] $statements
     */
    public function __construct(array $statements)
```

this is also fixed. No need to change the rector stuff.

You could even add both annotations if required:
```

    /**
     * @param Statement[] $statements
     * @psalm-param Statement[] $statements
     */

```

**Comment by martin-helmich** [comment]
Closing this for now; this is fixed in typo3-rector (thanks @sabbelasichon 🫶), and I've also merged #101, since that won't hurt either way.
